src/lib.rs

```diff
@@ -101,9 +104,9 @@ impl Tracer {
     /// # Arguments
     ///
     /// * `filename` - The filename in which to store trace packets.
-    pub fn trace_filename(mut self, filename: &str) -> Self {
-        self.trace_filename = CString::new(filename).expect("bad trace filename");
-        self
+    pub fn trace_filename(mut self, filename: &str) -> Result<Self, TraceMeError> {
```

Comment:
Is there a particular reason why `trace_filename` returns `Self`? Is it part of a "fluent" interface or ...?

Reply:
I'm not sure what a fluent interface is, but I was following the rust paradigm where you construct and configure an object in a chain:

```
let mything = Thing().setting1("blah").setting2("flibble");
```

My next PR will simplify this particular piece of code, so there is not possibility of failure in the chain.

Reply:
Looks like I just described a fluent interface :) Yes, a fluent interface, that's what I meant.

Reply:
OK. I just wanted to check you did it deliberately :)